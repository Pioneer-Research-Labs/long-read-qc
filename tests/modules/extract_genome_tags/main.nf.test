nextflow_process {

    name "Test Process extract_genome_tags"
    script "modules/extract_genome_tags/main.nf"
    process "extract_genome_tags"

    test("Should extract two genome tags from inputs") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.fromList([ tuple( [id:'WJ4Y3Y_MNT_008',
                genome:''],
                file('${projectDir}/test_data/WJ4Y3Y_MNT_008/WJ4Y3Y_samples.fasta'),
                file('${projectDir}/test_data/c.00432.dna'),
                file('${projectDir}/test_data/WJ4Y3Y_MNT_008/genome_tag_flanking.gb'))])
                """
            }
        }

        then {
            assert process.success
            assert path("$outputDir/WJ4Y3Y_MNT_008/extracted_genome_tags.tsv").readLines().size() == 2
            assert snapshot(
                ("$outputDir/WJ4Y3Y_MNT_008/extracted_genome_tags.tsv"),
                ("$outputDir/WJ4Y3Y_MNT_008/genome_tags.fasta"),
                ("$outputDir/WJ4Y3Y_MNT_008/cutadapt_genome_tags_info.tsv"),
                ("$outputDir/WJ4Y3Y_MNT_008/cutadapt_genome_tags_report.json")
            ).match()

        }

    }

}
