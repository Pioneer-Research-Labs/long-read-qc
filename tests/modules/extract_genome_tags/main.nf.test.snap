{
    "Should extract two genome tags from inputs": {
        "content": [
            "extracted_genome_tags.tsv:md5,9766e52d6aaf6266931df85580c3caa0",
            "genome_tags.fasta:md5,2fae16d07bc950eb02d18f03a2dcba3f",
            "cutadapt_genome_tags_info.tsv:md5,c804dc0c1d53a3a766c77037afd30c2c",
            "cutadapt_genome_tags_report.json:md5,7b2b08f248752223ca84bba9dadc7320"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-19T09:24:05.25498"
    }
}